### ISSUE TITLE
Database cannont be constructed from migrations

### ISSUE THREAD

**Issue by fathergoose** [gatorjuice/serviceweb_app#3]
### Problem
When running `rake db:migrate` an error is thrown that 'relation "users" does not exist'.

I believe [this commit](https://github.com/gatorjuice/serviceweb_app/commit/e409e37d5dbda1c5fa68ec36c97a5cbe3e78f4fb#diff-d77d5e503ad1439f585ac494268b351b) is the source of the trouble.

### Two Possible Solutions
- Carefuly reorder the contents of the squashed migration so that things are only referenced which already exist in the db
- Delete all migrations,  then generate tables one migration at a time

Let me know what you think. @gatorjuice, it may be best if you walk your way through this, since you are the most familiar with the schema. I will keep trying though, let you know if I feel good about a solution. We could pair up sometime too and try to work this out. Either way, Im a bit stuck on contributing since I cannot get the app to run yet;)

**Comment by gatorjuice** [comment]
Ah yeah your gonna just need to do rake db:schema:load instead of running
migrations.

On Saturday, March 12, 2016, Al Ilseman <notifications@github.com> wrote:

> Problem
>
> When running rake db:migrate an error is thrown that 'relation "users"
> does not exist'.
>
> I believe this commit
> <https://github.com/gatorjuice/serviceweb_app/commit/e409e37d5dbda1c5fa68ec36c97a5cbe3e78f4fb#diff-d77d5e503ad1439f585ac494268b351b>
> is the source of the trouble.
> Two Possible Solutions
>
>    - Carefuly reorder the contents of the squashed migration so that
>    things are only referenced which already exist in the db
>    - Delete all migrations, then generate tables one migration at a time
>
> Let me know what you think. @gatorjuice <https://github.com/gatorjuice>,
> it may be best if you walk your way through this, since you are the most
> familiar with the schema. I will keep trying though, let you know if I feel
> good about a solution. We could pair up sometime too and try to work this
> out.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gatorjuice/serviceweb_app/issues/3>.
>


-- 
Sent by Jamie's pocket computer

**Comment by fathergoose** [comment]
Oh, I see. That worked.
I'm going to see if I can get it to work with migrations too, I believe this is the "sandard" way to have it work. Let me know if you have any thoughts/tips here.

**Comment by fathergoose** [comment]
Maybe you have seen [this](http://stackoverflow.com/questions/5224827/generate-a-migration-file-from-schema-rb)?

**Comment by gatorjuice** [comment]
Sounds good dude. Thanks again. I know exactly why it won't let you
migrate. I did a roll back once and it didn't delete the migration history.
So I think there is a migration file that says to change the name of one of
the models but the old name isn't there as the migrations are being stepped
through.

In other news, the rails recommended way is in fact to run rake
db:schema:load I just noticed once that it tells you that in the schema
file comments up at the top. I would have never known otherwise. It would
be great to be able to run it either way though

On Sunday, March 13, 2016, Al Ilseman <notifications@github.com> wrote:

> Oh, I see. That worked.
> I'm going to see if I can get it to work with migrations too, I believe
> this is the "sandard" way to have it work. Let me know if you have any
> thoughts/tips here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gatorjuice/serviceweb_app/issues/3#issuecomment-195965710>
> .
>


-- 
Sent by Jamie's pocket computer
